src/main/java/prgms/vouchermanagementapp/controller/VoucherManagementController.java

```diff
@@ -0,0 +1,37 @@
+package prgms.vouchermanagementapp.controller;
+
+import prgms.vouchermanagementapp.io.MenuType;
+import prgms.vouchermanagementapp.io.Reader;
+import prgms.vouchermanagementapp.io.Writer;
+import prgms.vouchermanagementapp.io.message.SystemMessage;
+
+public class VoucherManagementController implements Runnable{
+    private final Reader reader;
+    private final Writer writer;
+    private final RunningState state;
+
+    public VoucherManagementController(Reader reader, Writer writer, RunningState state) {
+        this.reader = reader;
+        this.writer = writer;
+        this.state = state;
+    }
+
+    @Override
+    public void run() {
+        while(state.isRunning()){
+            writer.print(MenuType.getMessages());
```

Comment:
```suggestion
            writer.printMenuMessage()
```
여기도 마찬가지에요. 객체에게 메시지를 보내봐요

Reply:
@hongbin-dev 

[Refactor: 기존의 Controller가 갖는 '어떤 메시지를 출력할 것인지 정하는 책임'을 Writer에게 위임](https://github.com/prgrms-be-devcourse/springboot-basic/pull/528/commits/f5e1607189c7112c7ad13f2c2222a86e97e039ef)

- 피드백을 적용하여 Controller 객체가 Writer 객체에게 메시지를 보내도록 수정했습니다.
- ❓ 커밋에 작성한 것처럼 수정하다 보면 모든 메뉴 타입에 대응하는 메서드를 하나씩 만들어야 될텐데 이 부분은 '불필요한 코드의 중복' 인지, 아니면 'SRP'를 잘 지키는 방법인지 궁금합니다.
- ❓ 더 나은 방법으로 개선시킨다면 어떤 방법이 있을 지 궁금합니다!

Reply:
앗 이 부분은 제가 컨텍스트를 놓쳐 이해를 못했군요. 선호님이 작성해주신 방향으로 가는게 일반적인 것 같습니다.

다만 `MenuType.getMessages()`이 부분이 단순 출력을 위한 메서드가 아닌가? 생각이 들었습니다. print()에 기존에 제공하는`MenuType.values()`를 넘기고 출력객체에서 포맷을 설정하는건 어떨까요?